### ISSUE TITLE
Syslog, Libdoc, Testdoc and Tidy don't create directory for outputs

### ISSUE THREAD

**Issue by bkhouri** [robotframework/robotframework#2767]
# Summary
If the `ROBOT_SYSLOG_FILE` environment variable is set to a directory and filename, `robot` will fails to create the System log file if the directory does not exist.

# Environment:

- Robot Framework version 3.0.2
- Python version 2.7.10
- OS: Mac OS Sierra version 10.12.6 

# Reproducible
100%

# Steps to Reproduce:
1. Create the following dummy test case named `Foo.robot` 
    ```robot
    *** Test Cases ***
    Foo Test Case
        Log     just logging
    ```
1. Run the following command `ROBOT_SYSLOG_FILE=./garbage/syslog.log robot  Foo.robot`
    ```
    $ ROBOT_SYSLOG_FILE=./garbage/syslog.log robot  Foo.robot
    [ ERROR ] Opening syslog file './garbage/syslog.log' failed: No such file or directory
    [ ERROR ] Opening syslog file './garbage/syslog.log' failed: No such file or directory
    ==============================================================================
    Foo                                                                           
    ==============================================================================
    Foo Test Case                                                         | PASS |
    ------------------------------------------------------------------------------
    Foo                                                                   | PASS |
    1 critical test, 1 passed, 0 failed
    1 test total, 1 passed, 0 failed
    ==============================================================================
    Output:  /Users/******/output.xml
    Log:     /Users/******/log.html
    Report:  /Users/******/report.html
    ```
 
Robot framework should create the directory before attempting to get a filehandler on the SYSLOG file.

**Comment by pekkaklarck** [comment]
I guess this would make sense, especially when we already do it with the normal output files. Interested to provide a PR? It would still have a change to make it into RF 3.0.3 planned to be released this week.

**Comment by rasjani** [comment]
Similar issue also happens in libdoc when trying to create the keyword documentation into a directory that does not exist here: https://github.com/robotframework/robotframework/blob/master/src/robot/utils/robotio.py#L23

**Comment by pekkaklarck** [comment]
OK. I'll add this to RF 3.1.2 scope and we can get both of these fixed.

**Comment by pekkaklarck** [comment]
This is a pretty trivial fix but RF 3.1.2 is also late. Postponing to 3.2.

**Comment by pekkaklarck** [comment]
This was fixed by PR #3186 by @mihaiparvu.

**Comment by pekkaklarck** [comment]
Also Tidy and Testdoc were affected by this issue.
